### PR TITLE
feat(credits): add credits for each group

### PR DESCRIPTION
**Pull request by samueldurantes**
fix: https://github.com/vinibgoulart/wpp-playground/issues/11

### CODE REVIEW

**Review by vinibgoulart — APPROVED** [review]
nice!

File: src/wweb/middleware/consumerCredits.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  if (!group) {
+    return {
+      error: 'Group not found',
+    };
+  }
+
+  const creditsDiscount = getCreditsDiscountPerCommand(command);
+
+  if (group.credits < creditsDiscount) {
```
Comment (daniloab) [inline src/wweb/middleware/consumerCredits.ts:38]: some points here:
- how to handle concurrency, two users write at the same time
- how to debit the correct message and only one time
- this must be idempotent
- creditsDiscount? what does this mean? handle this like an account with balance and debits

**Comment by Grubba27** [comment]
Should be a command so that the group knows how many credits they still have left

**Comment by vinibgoulart** [comment]
good catch
https://github.com/vinibgoulart/wpp-playground/issues/27

File: src/wweb/middleware/consumerCredits.ts (hunk 1 above)
Comment (daniloab) [reply src/wweb/middleware/consumerCredits.ts:38]: By start could have less validations

**Pull request merged**